apps/site/assets/ts/schedule/components/RapidTransitHoursOfOperation.tsx

```diff
@@ -0,0 +1,126 @@
+import { parseISO } from "date-fns";
+import { formatInTimeZone } from "date-fns-tz";
+
+import { map, uniqueId, sortBy, filter } from "lodash";
+import React, { ReactElement } from "react";
+import ExpandableBlock from "../../components/ExpandableBlock";
+import useHoursOfOperation from "../../hooks/useHoursOfOperation";
+import useUrlSearchParams from "../../hooks/useUrlSearchParams";
+import { EnhancedRoute, StopHours } from "../../__v3api";
+import pdfLink from "../helpers/hoursOfOperationHelpers";
+import { SchedulePDF } from "./__schedule";
+
+const SCHEDULE_PARAM = "schedule_direction[direction_id]";
+
+const parseOutTime = (dateTimeString: string): string => {
+  const dateTime = parseISO(dateTimeString);
+  return formatInTimeZone(dateTime, "America/New_York", "h:mmaaa");
+};
+
+const getSchedule = (
+  dataArray: StopHours[][] | StopHours[],
+  currentDirection: number
+): ReactElement<HTMLElement>[] | ReactElement<HTMLElement> => {
+  const filteredData = filter(
+    dataArray[currentDirection],
+    (stopData: StopHours) => stopData.is_terminus
+  );
+  const sortedData = sortBy(
+    filteredData,
+    (stopData: StopHours) => stopData.stop_name
+  );
+  const mappedData = map(sortedData, (stopData: StopHours) => (
+    <div key={uniqueId()} className="fs-18">
+      <span className="pe-16">{stopData.stop_name}</span>
+      <span className="font-weight-bold">
+        {`${parseOutTime(stopData.first_departure)} - ${parseOutTime(
+          stopData.last_departure
+        )}`}
+      </span>
+    </div>
+  ));
+
+  return mappedData;
+};
+
+const regularScheduleHTML = (): ReactElement<HTMLElement> => (
+  <div className="font-weight-bold fs-12 pb-14">Regular schedule</div>
+);
+
+const trainsEveryHTML = (
+  minuteString: string,
+  showExceptDuringRushHour: boolean
+): ReactElement<HTMLElement> => {
+  const rushHour = showExceptDuringRushHour ? " except during rush hour" : "";
+  const trainsEvery = `Trains every ${minuteString} minutes ${rushHour}`;
+  return <div className="fs-14 pt-14">{trainsEvery}</div>;
+};
+
+const RapidTransitHoursOfOperation = ({
+  route,
+  pdfs
+}: {
+  route: EnhancedRoute;
+  pdfs: SchedulePDF[];
+}): ReactElement<HTMLElement> => {
+  const hours = useHoursOfOperation(route.id);
+  const urlParams = useUrlSearchParams();
+
+  const currentDirectionParam = urlParams?.get(SCHEDULE_PARAM);
+  const currentDirection = currentDirectionParam
+    ? Number(currentDirectionParam)
+    : 0;
```

Comment:
I think the the best solution would be to allow passing params to the endpoint.  That way it would only return the data required by the page instead of all of it.  Something like `direction_id=0&only_terminus`.  Maybe a sort key as well?
Do we have a pattern to follow for passing query params to the web server?

Reply:
I'm not really sure I follow your question, but you can use `URLSearchParams` to set URL parameters as well, then output the whole thing via `.toString()`.

Reply:
It was a question about sending query params to the `/hours` end point.  I don't think its needed any more, but I wanted to know if there was a convention about how we name query params

Reply:
ah thanks for clarifying for me -- we don't have any convention as of now!